In [1]:
# Standard data science libraries
import pandas as pd
import numpy as np
from pathlib import Path

# Display all cell outputs
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = 'all'

import plotly
# Authenticate with your account
plotly.tools.set_credentials_file(username='####',api_key='###')

import plotly.plotly as py
import plotly.graph_objs as go

# Offline mode
from plotly.offline import iplot
plotly.offline.init_notebook_mode(connected=True)

In [2]:
from bnb.data_processing.read import read_csvfile
from bnb.data_processing.clean import group_by_date
from bnb.data_processing.clean import price_format
from bnb.data_processing.clean import bathrooms_int
from bnb.data_processing.clean import shared_bathroom
from bnb.data_processing.clean import super_host
from bnb.data_processing.clean import beds_int
from bnb.data_processing.clean import num_of_stars

## Reading Data

In [3]:
Data_dir = Path('C:/Users/KBrig/AirBnBRepo/AirBnB/data')
Data_dir

WindowsPath('C:/Users/KBrig/AirBnBRepo/AirBnB/data')

In [4]:
NewYork_CSV = Data_dir/'AirBnBNY.csv'
Bangkok_CSV = Data_dir/'AirBnbDataBangkok.csv'
Cairo_CSV = Data_dir/'AirBnbDataCairo.csv'


## NewYork

In [5]:
NewYorkData = read_csvfile(NewYork_CSV)
NewYorkData.head()


Unnamed: 0       Accuracy_stars  \
0           0    Rating 5 out of 5   
1           1  Rating 4.5 out of 5   
2           2    Rating 5 out of 5   
3           3    Rating 5 out of 5   
4           4    Rating 5 out of 5   

                                           Amenities     Bathrooms_num  \
0  ['Indoor fireplace', 'Wifi', 'Iron', 'Laptop f...            1 bath   
1  ['Indoor fireplace', 'Wifi', 'Laptop friendly ...     1 shared bath   
2  ['Wifi', 'Essentials', 'Heating', 'Air conditi...     1 shared bath   
3  ['Wifi', 'Iron', 'Laptop friendly workspace', ...    2 shared baths   
4  ['Wifi', 'Washer', 'Dryer', 'Iron', 'Laptop fr...  1.5 shared baths   

  Bedrooms_no Beds_no              CheckIn        CheckIn_stars  \
0   1 bedroom   1 bed  2019-03-21 00:00:00    Rating 5 out of 5   
1   1 bedroom   1 bed  2019-03-21 00:00:00  Rating 4.5 out of 5   
2   1 bedroom   1 bed  2019-03-21 00:00:00    Rating 5 out of 5   
3   1 bedroom   1 bed  2019-03-21 00:00:00    Rating 5 out of 5   
4   1 bedroom   1 bed  2019-03-21 00:00:00    Rating 5 out of 5   

              CheckOut  Cleanliness_stars         ...           Latitude  \
0  2019-03-26 00:00:00  Rating 5 out of 5         ...           40.73321   
1  2019-03-26 00:00:00  Rating 4 out of 5         ...           40.76580   
2  2019-03-26 00:00:00  Rating 5 out of 5         ...           40.75972   
3  2019-03-26 00:00:00  Rating 5 out of 5         ...           40.81386   
4  2019-03-26 00:00:00  Rating 5 out of 5         ...           40.69284   

        Location_stars  Longitude  Price  \
0    Rating 5 out of 5  -74.00242   $244   
1    Rating 5 out of 5  -73.95812    $62   
2  Rating 4.5 out of 5  -74.03827    $44   
3    Rating 5 out of 5  -73.95936    $78   
4  Rating 4.5 out of 5  -73.94418    $80   

                                              Review  Reviewers_no  \
0  87 Reviews\r\nAccuracy\r\nCommunication\r\nCle...           NaN   
1  5 Reviews\r\nAccuracy\r\nCommunication\r\nClea...           NaN   
2  83 Reviews\r\nAccuracy\r\nCommunication\r\nCle...           NaN   
3  32 Reviews\r\nAccuracy\r\nCommunication\r\nCle...           NaN   
4  28 Reviews\r\nAccuracy\r\nCommunication\r\nCle...           NaN   

          Sleeping_engagment              Super_host  \
0   Bedroom 1\r\n1 queen bed  Brian is a Superhost ·   
1  Bedroom 1\r\n1 single bed    Samet is a Superhost   
2   Bedroom 1\r\n1 queen bed  Rebecca is a Superhost   
3   Bedroom 1\r\n1 queen bed    James is a Superhost   
4                        NaN      Cat is a Superhost   

                                               Title          Value_stars  
0  *Luxury 1 Bedrm Private Designer Home & Roofdeck*  Rating 4.5 out of 5  
1        Single Bedroom-Upper East Side-Central Park    Rating 4 out of 5  
2  Quiet room for sleeping - 15 min from Times Sq...    Rating 5 out of 5  
3        Cozy room C in upper west side NY, Columbia    Rating 5 out of 5  
4                                    Cozy in Bedstuy    Rating 5 out of 5  

[5 rows x 23 columns]

## Bangkok

In [6]:
BangkokData = read_csvfile(Bangkok_CSV)
BangkokData.head()

CheckIn         CheckOut  Has_Weekend  \
0  2019-03-12 0:00  2019-03-17 0:00         True   
1  2019-03-12 0:00  2019-03-17 0:00         True   
2  2019-03-12 0:00  2019-03-17 0:00         True   
3  2019-03-12 0:00  2019-03-17 0:00         True   
4  2019-03-12 0:00  2019-03-17 0:00         True   

                                               Title Price  \
0                Cosy room with 3 mins walk from MRT   $34   
1                                       曼谷市区BTS旁品质公寓   $48   
2  Modern apartment close to city near sky train(...   $27   
3  Outstanding Location in Heart of Bangkok 2mins...   $35   
4  5 Min walk MRT Phraram9, 1stop RodFai Train Ma...   $34   

                 Super_host  Latitude  Longitude    EntireHomeVsRoom  \
0  Thongjuta is a Superhost  13.78165  100.57453    ENTIRE APARTMENT   
1                       NaN  13.69750  100.60708    ENTIRE APARTMENT   
2       Kiek is a Superhost  13.71252  100.46783  ENTIRE CONDOMINIUM   
3                       NaN  13.72032  100.49623  ENTIRE CONDOMINIUM   
4      Patty is a Superhost  13.75738  100.56318    ENTIRE APARTMENT   

  Bedrooms_no                        ...                          \
0   1 bedroom                        ...                           
1   1 bedroom                        ...                           
2   1 bedroom                        ...                           
3   1 bedroom                        ...                           
4         NaN                        ...                           

                              Sleeping_engagment  \
0                         Bedroom 1\n1 queen bed   
1                        Bedroom 1\n1 double bed   
2                         Bedroom 1\n1 queen bed   
3                          Bedroom 1\n1 king bed   
4  Common spaces\n2 queen beds, 1 floor mattress   

                                           Amenities     Accuracy_stars  \
0  ['Wifi', 'Iron', 'Cable TV', 'Laptop friendly ...  Rating 5 out of 5   
1  ['Wifi', 'Indoor fireplace', 'Heating', 'Iron'...                NaN   
2  ['Wifi', 'Heating', 'Laptop friendly workspace...  Rating 5 out of 5   
3  ['Wifi', 'Iron', 'Cable TV', 'Laptop friendly ...  Rating 5 out of 5   
4  ['Wifi', 'Iron', 'Laptop friendly workspace', ...                NaN   

  Communication_stars    Cleanliness_stars     Location_stars  \
0   Rating 5 out of 5  Rating 4.5 out of 5  Rating 5 out of 5   
1                 NaN                  NaN                NaN   
2   Rating 5 out of 5    Rating 5 out of 5  Rating 4 out of 5   
3   Rating 5 out of 5  Rating 4.5 out of 5  Rating 5 out of 5   
4                 NaN                  NaN                NaN   

       CheckIn_stars          Value_stars Reviewers_no  \
0  Rating 5 out of 5    Rating 5 out of 5          NaN   
1                NaN                  NaN          NaN   
2  Rating 5 out of 5  Rating 4.5 out of 5          NaN   
3  Rating 5 out of 5    Rating 5 out of 5          NaN   
4                NaN                  NaN          NaN   

                                              Review  
0  101 Reviews\nAccuracy\nCommunication\nCleanlin...  
1  No reviews (yet)\nBe one of the first guests t...  
2  4 Reviews\nAccuracy\nCommunication\nCleanlines...  
3  3 Reviews\nAccuracy\nCommunication\nCleanlines...  
4  No reviews (yet)\nThis host has 1631 reviews f...  

[5 rows x 22 columns]

## Cairo

In [7]:
CairoData = read_csvfile(Cairo_CSV)
CairoData.head()

CheckIn         CheckOut  Has_Weekend  \
0  2019-03-14 0:00  2019-03-19 0:00         True   
1  2019-03-14 0:00  2019-03-19 0:00         True   
2  2019-03-14 0:00  2019-03-19 0:00         True   
3  2019-03-14 0:00  2019-03-19 0:00         True   
4  2019-03-14 0:00  2019-03-19 0:00         True   

                                               Title Price Super_host  \
0                                         Dad Haus!x   $15        NaN   
1                    Apartment in Rehab City, Cairo.   $12        NaN   
2  Sunny and friendly apartment close to the airport   $22        NaN   
3  Cozy 2 persons budget room in luxurious/safe area   $19        NaN   
4         Girls apartment - Sunny and calm apartment   $23        NaN   

   Latitude  Longitude             EntireHomeVsRoom Bedrooms_no  \
0  29.95961   31.25858    Private room in apartment   1 bedroom   
1  30.05760   31.48779             Entire apartment   1 bedroom   
2  30.06085   31.49540    Private room in apartment   1 bedroom   
3  30.06284   31.51743  Private room in condominium   1 bedroom   
4  30.06263   31.48585    Private room in apartment   1 bedroom   

                         ...                          \
0                        ...                           
1                        ...                           
2                        ...                           
3                        ...                           
4                        ...                           

         Sleeping_engagment  \
0   Bedroom 1\n2 queen beds   
1    Bedroom 1\n1 queen bed   
2                       NaN   
3  Bedroom 1\n2 single beds   
4    Bedroom 1\n1 queen bed   

                                           Amenities     Accuracy_stars  \
0  ['Wifi', 'Indoor fireplace', 'Dryer', 'Laptop ...                NaN   
1  ['Wifi', 'Laptop friendly workspace', 'Washer'...                NaN   
2  ['Wifi', 'Indoor fireplace', 'Cable TV', 'Lapt...  Rating 4 out of 5   
3  ['Wifi', 'Laptop friendly workspace', 'Iron', ...  Rating 5 out of 5   
4  ['Wifi', 'Cable TV', 'Laptop friendly workspac...                NaN   

  Communication_stars    Cleanliness_stars       Location_stars  \
0                 NaN                  NaN                  NaN   
1                 NaN                  NaN                  NaN   
2   Rating 5 out of 5  Rating 3.5 out of 5  Rating 4.5 out of 5   
3   Rating 5 out of 5    Rating 5 out of 5    Rating 5 out of 5   
4                 NaN                  NaN                  NaN   

       CheckIn_stars        Value_stars Reviewers_no  \
0                NaN                NaN          NaN   
1                NaN                NaN          NaN   
2  Rating 5 out of 5  Rating 4 out of 5          NaN   
3  Rating 5 out of 5  Rating 5 out of 5          NaN   
4                NaN                NaN          NaN   

                                              Review  
0  No reviews (yet)\nBe one of the first guests t...  
1  No reviews (yet)\nThis host has 11 reviews for...  
2  6 Reviews\nAccuracy\nCommunication\nCleanlines...  
3  4 Reviews\nAccuracy\nCommunication\nCleanlines...  
4  2 Reviews\nAmelia\nDecember 2018\nThank you ve...  

[5 rows x 22 columns]

## Data Preprocessing

## NewYork

In [8]:
NewYorkData.shape

(3957, 23)

In [9]:
NewYorkData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3957 entries, 0 to 3956
Data columns (total 23 columns):
Unnamed: 0             3957 non-null int64
Accuracy_stars         3614 non-null object
Amenities              3957 non-null object
Bathrooms_num          3922 non-null object
Bedrooms_no            3348 non-null object
Beds_no                3895 non-null object
CheckIn                3957 non-null object
CheckIn_stars          3614 non-null object
CheckOut               3957 non-null object
Cleanliness_stars      3614 non-null object
Communication_stars    3614 non-null object
EntireHomeVsRoom       3957 non-null object
Has_Weekend            3957 non-null bool
Latitude               3957 non-null float64
Location_stars         3614 non-null object
Longitude              3957 non-null float64
Price                  3917 non-null object
Review                 3956 non-null object
Reviewers_no           0 non-null float64
Sleeping_engagment     3158 non-null object
Super_host      

We have a total of 23 features as shown above: 1 boolean (Has_Weekend), 3 numeric, float64(Latitude, Longitude, Reviewers_num), and the rest are saved as data objects. Let's go through them!! 



So, we need to keep ONLY features that can add information to our analysis. Predictors with many null values can be considered redundant and ignore tham.

### Missing Data

In [10]:
NewYorkData.isnull().sum()

Unnamed: 0                0
Accuracy_stars          343
Amenities                 0
Bathrooms_num            35
Bedrooms_no             609
Beds_no                  62
CheckIn                   0
CheckIn_stars           343
CheckOut                  0
Cleanliness_stars       343
Communication_stars     343
EntireHomeVsRoom          0
Has_Weekend               0
Latitude                  0
Location_stars          343
Longitude                 0
Price                    40
Review                    1
Reviewers_no           3957
Sleeping_engagment      799
Super_host             2010
Title                     0
Value_stars             343
dtype: int64

In [11]:
## Features with large number of missing data
[column for column in NewYorkData.columns if NewYorkData[column].count() / len(NewYorkData) <= 0.7]

['Reviewers_no', 'Super_host']

So "Super_host" can have two possible values: Yes/True or No/False. Samples with null values are actually representing non super_host. On the other hand, having 3957 of null values for the Reviews_no says that there is something wrong with collecting data for this variable. So, it is better to exclude it.

Also, for the reviews' features, we see that we have the same number of null values for the Accuracy,CheckIn,Cleanliness,Communication, Location and Value Stars. This make sense since this number points to the homes with "no reviews yet".  We are going to have this invistigation for all features!

Let's classify the features: numeric, catogerical,etc.


So we have 23 variables six of them are categorical variables reflecting the clients' satisfaction in different aspects: 'Accuracy_stars','CheckIn_stars','Cleanliness_stars','Communication_stars','Location_stars','Value_stars'.

Other categorical variables such as 'EntireHomeVsRoom','Has_Weekend', and 'Sleeping_engagment'.

We also have dates: 'CheckIn','CheckOut'.

We have three numeric features: 'Price','Bathrooms_num', 'Bedrooms_no', 'Beds_no', 'Latitude','Longitude'.

rest:'Review', 'Title', and 'Super_host'


Let's tidy up!


In [12]:
NewYorkData.drop(['Unnamed: 0','Reviewers_no'],axis=1, inplace=True)

### Does it have a weekend?

We have originally one boolean variable "Has_Weekend". Let's check its levels!

In [13]:
NewYorkData['Has_Weekend'].unique()

array([ True])

So "Has_Weekend" doesn't say more than every time_period is a mix of weekdays and weekends. Thus, we can ignore it!

In [14]:
NewYorkData.drop('Has_Weekend',axis=1, inplace=True)

### Is Super Host?

In [15]:
NewYorkData.Super_host[0:10]

0    Brian is a Superhost ·
1      Samet is a Superhost
2    Rebecca is a Superhost
3      James is a Superhost
4        Cat is a Superhost
5       Rick is a Superhost
6                       NaN
7    Shailah is a Superhost
8                       NaN
9       Obed is a Superhost
Name: Super_host, dtype: object

In [16]:
NewYorkData['Super_host'] = super_host(NewYorkData)

True = 1, False = 0

In [17]:
NewYorkData['Super_host'][0:10]

0    1
1    1
2    1
3    1
4    1
5    1
6    0
7    1
8    0
9    1
Name: Super_host, dtype: int64

## Numeric Data

### Price

In [18]:
print(type(NewYorkData['Price'][0]))
print(NewYorkData['Price'][0:10])

<class 'str'>
0    $244
1     $62
2     $44
3     $78
4     $80
5    $116
6     $96
7    $119
8     $73
9     $63
Name: Price, dtype: object


In [19]:
NewYorkData['Price'] = price_format(NewYorkData)

In [20]:
print(type(NewYorkData['Price'][0]))
print(NewYorkData['Price'][0:10])

<class 'numpy.float64'>
0    244.0
1     62.0
2     44.0
3     78.0
4     80.0
5    116.0
6     96.0
7    119.0
8     73.0
9     63.0
Name: Price, dtype: float64


### Number of Bathrooms

In [21]:
print(type(NewYorkData['Bathrooms_num'][0]))
print(NewYorkData['Bathrooms_num'].unique())

<class 'str'>
['1 bath' '1 shared bath' '2 shared baths' '1.5 shared baths'
 '1.5 private baths' '1 private bath' '2 private baths' nan
 '2.5 shared baths' '2 baths' '2.5 private baths' '3 shared baths'
 '4 shared baths' '0 baths' 'Half-bath' 'Shared half-bath' '1.5 baths'
 '2.5 baths' '3 baths' '0 shared baths' '3.5 shared baths']


In [146]:
NewYorkData['Bathrooms_num_int']= float(0)

In [147]:
NewYorkData['Bathrooms_num_int'] = bathrooms_int(NewYorkData)

c:\users\kbrig\airbnbrepo\airbnb\bnb\data_processing\clean.py:15: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy



In [150]:
for i in range(10):
    print(NewYorkData['Bathrooms_num'][i],NewYorkData['Bathrooms_num_int'][i])

1 bath 1.0
1 shared bath 1.0
1 shared bath 1.0
2 shared baths 2.0
1.5 shared baths 1.5
1 shared bath 1.0
1 shared bath 1.0
1 shared bath 1.0
1 shared bath 1.0
2 shared baths 2.0


### Shared vs. Private Bathroom

In [151]:
NewYorkData['shared_bathroom']= shared_bathroom(NewYorkData)

In [152]:
NewYorkData[['Bathrooms_num','shared_bathroom']][0:10]

Bathrooms_num  shared_bathroom
0            1 bath            False
1     1 shared bath             True
2     1 shared bath             True
3    2 shared baths             True
4  1.5 shared baths             True
5     1 shared bath             True
6     1 shared bath             True
7     1 shared bath             True
8     1 shared bath             True
9    2 shared baths             True

### Number of Bedrooms

In [23]:
print(type(NewYorkData['Bedrooms_no'][0]))
print(NewYorkData['Bedrooms_no'].unique())

<class 'str'>
['1 bedroom' 'Super cute 1 bedroom apartment close to TimeSquare' nan
 'Large private bedroom 1 stop from Manhattan'
 '1 large bedroom HARLEM/NYC/terrace/'
 'Cozy tiny bedroom w/large living-room & kitchen.' '2 bedrooms'
 'Beautiful bedroom close to NYC' '3 bedrooms'
 'Spacious 3 bedroom apartment in WILLIAMSBURG'
 'Urban Farmers bedroom complete with fort' '4 bedrooms'
 'Cozy bedroom in townhouse with backyard in LIC'
 'Convenient Private bedroom 10 MIN FROM NYC'
 'Cozy bedroom with balcony. 25min TimeSquare'
 'Sunny, Clean, Nice bedroom available in Bushwick'
 'Large bedroom/Heart of Brooklyn'
 'Private bedroom in the Upper West Side, Manhattan'
 'Large queen size bedroom in huge loft apt'
 'Nice bedroom in east village apartment'
 'Very large and comfortable Manhattan one bedroom.'
 'Oasis bedroom for solo travelers' 'Big bedroom with queen size Bed'
 'Cozy bedroom by Yankee Stadium'
 'Private bedroom&bathroom in the heart of Manhattan'
 'Huge private bedroom for quiet

### Number of Beds

In [24]:
print(type(NewYorkData['Beds_no'][0]))
print(NewYorkData['Beds_no'].unique())

<class 'str'>
['1 bed' '0 beds' 'Obed' '2 beds' 'PRIVATE ROOM IN BED AND BREAKFAST'
 '3 beds' nan '5 beds' '4 beds' '8 beds'
 'SHARED ROOM IN BED AND BREAKFAST' '6 beds' 'ENTIRE BED AND BREAKFAST']


In [25]:
NewYorkData['Beds_int'] = float(0)

In [26]:
NewYorkData['Beds_int'] = beds_int(NewYorkData)

c:\users\kbrig\airbnbrepo\airbnb\bnb\data_processing\clean.py:29: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy

c:\users\kbrig\airbnbrepo\airbnb\bnb\data_processing\clean.py:27: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy

c:\users\kbrig\airbnbrepo\airbnb\bnb\data_processing\clean.py:25: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy

c:\users\kbrig\airbnbrepo\airbnb\bnb\data_processing\clean.py:23: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFr

In [27]:
for i in range(10):
    print(NewYorkData['Beds_no'][i],NewYorkData['Beds_int'][i])

1 bed 1.0
1 bed 1.0
1 bed 1.0
1 bed 1.0
1 bed 1.0
1 bed 1.0
1 bed 1.0
0 beds 0.0
1 bed 1.0
Obed 0.0


### Latitude

In [28]:
print(type(NewYorkData['Latitude'][0]))
print(NewYorkData['Latitude'].unique())

<class 'numpy.float64'>
[40.73321 40.7658  40.75972 ... 40.77264 40.76406 40.73354]


### Longitude

In [29]:
print(type(NewYorkData['Longitude'][0]))
print(NewYorkData['Longitude'].unique())

<class 'numpy.float64'>
[-74.00242 -73.95812 -74.03827 ... -74.03762 -74.02755 -73.90769]


Thanks god! we have Latitude and Longitude ready to go!

### Bangkok

In [30]:
BangkokData.columns

Index(['CheckIn', 'CheckOut', 'Has_Weekend', 'Title', 'Price', 'Super_host',
       'Latitude', 'Longitude', 'EntireHomeVsRoom', 'Bedrooms_no', 'Beds_no',
       'Bathrooms_num', 'Sleeping_engagment', 'Amenities', 'Accuracy_stars',
       'Communication_stars', 'Cleanliness_stars', 'Location_stars',
       'CheckIn_stars', 'Value_stars', 'Reviewers_no', 'Review'],
      dtype='object')

### Does it have a weekend?

In [31]:
BangkokData['Has_Weekend'].unique()

array([ True])

In [32]:
BangkokData.drop(['Has_Weekend','Reviewers_no'],axis=1, inplace=True)

### Is Super Host?

In [33]:
BangkokData.Super_host[0:10]

0     Thongjuta is a Superhost
1                          NaN
2          Kiek is a Superhost
3                          NaN
4         Patty is a Superhost
5         Nukul is a Superhost
6                          NaN
7                          NaN
8                          NaN
9    Kinifrog is a Superhost ·
Name: Super_host, dtype: object

In [34]:
BangkokData['Super_host'] = super_host(BangkokData)

In [35]:
BangkokData.Super_host[0:10]

0    1
1    0
2    1
3    0
4    1
5    1
6    0
7    0
8    0
9    1
Name: Super_host, dtype: int64

### Price

In [36]:
BangkokData.Price[0:10]

0    $34
1    $48
2    $27
3    $35
4    $34
5    $23
6    $27
7    $51
8    $38
9    $36
Name: Price, dtype: object

In [37]:
BangkokData['Price'] = price_format(BangkokData)

In [38]:
print(type(BangkokData['Price'][0]))
print(BangkokData['Price'][0:10])

<class 'numpy.float64'>
0    34.0
1    48.0
2    27.0
3    35.0
4    34.0
5    23.0
6    27.0
7    51.0
8    38.0
9    36.0
Name: Price, dtype: float64


### Number of Bathrooms

In [118]:
print(type(BangkokData['Bathrooms_num'][0]))
print(BangkokData['Bathrooms_num'].unique())

<class 'str'>
['1 bath' '1 private bath' '2 baths' '1.5 baths' '0 baths' '1 shared bath'
 '2 shared baths' '4 shared baths' '1.5 private baths' '8 shared baths'
 '1.5 shared baths' '3 baths' '4 baths' '2.5 baths' '3.5 baths' nan
 '4.5 baths'
 'Awesome place! Laundry service was available and affordable, a well equipped kitchen was provided, and the bathrooms and rooms were clean.'
 '6 shared baths' '4.5 shared baths']


In [119]:
BangkokData['Bathrooms_num_int']= float(0)

In [124]:
BangkokData[BangkokData['Bathrooms_num']== 'Awesome place! Laundry service was available and affordable, a well equipped kitchen was provided, and the bathrooms and rooms were clean.']

CheckIn         CheckOut  \
4028  2019-09-10 0:00  2019-09-15 0:00   
4330  2019-09-24 0:00  2019-09-29 0:00   
4609  2019-10-08 0:00  2019-10-13 0:00   
4919  2019-10-22 0:00  2019-10-27 0:00   
5202  2019-11-05 0:00  2019-11-10 0:00   
5207  2019-11-05 0:00  2019-11-10 0:00   
5519  2019-11-19 0:00  2019-11-24 0:00   
5786  2019-12-03 0:00  2019-12-08 0:00   
6079  2019-12-17 0:00  2019-12-22 0:00   
6338  2019-12-31 0:00  2020-01-05 0:00   
6697  2020-01-14 0:00  2020-01-19 0:00   
7296  2020-02-11 0:00  2020-02-16 0:00   
7604  2020-02-25 0:00  2020-03-01 0:00   

                                                  Title  Price  Super_host  \
4028  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
4330  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
4609  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
4919  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
5202  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
5207  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
5519  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
5786  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
6079  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
6338  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
6697  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
7296  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
7604  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   

      Latitude  Longitude       EntireHomeVsRoom Bedrooms_no Beds_no  \
4028    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
4330    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
4609    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
4919    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
5202    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
5207    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
5519    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
5786    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
6079    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
6338    13.728    100.527                    NaN   1 bedroom     NaN   
6697    13.728    100.527                    NaN   1 bedroom     NaN   
7296    13.728    100.527                    NaN   1 bedroom     NaN   
7604    13.728    100.527                    NaN   1 bedroom     NaN   

            ...             Sleeping_engagment  \
4028        ...         Bedroom 1\n3 bunk beds   
4330        ...         Bedroom 1\n3 bunk beds   
4609        ...         Bedroom 1\n3 bunk beds   
4919        ...         Bedroom 1\n3 bunk beds   
5202        ...         Bedroom 1\n3 bunk beds   
5207        ...         Bedroom 1\n3 bunk beds   
5519        ...         Bedroom 1\n3 bunk beds   
5786        ...         Bedroom 1\n3 bunk beds   
6079        ...         Bedroom 1\n3 bunk beds   
6338        ...         Bedroom 1\n3 bunk beds   
6697        ...         Bedroom 1\n3 bunk beds   
7296        ...         Bedroom 1\n3 bunk beds   
7604        ...         Bedroom 1\n3 bunk beds   

                                              Amenities     Accuracy_stars  \
4028  ['Wifi', 'Cable TV', 'Laptop friendly workspac...  Rating 5 out of 5   
4330  ['Wifi', 'Cable TV', 'Laptop friendly workspac...  Rating 5 out of 5   
4609  ['Wifi', 'Cable TV', 'Laptop friendly workspac...  Rating 5 out of 5   
4919  ['Wifi', 'Cable TV', 'Laptop friendly workspac...  Rating 5 out of 5   
5202  ['Wifi', 'Cable TV', 'Laptop friendly workspac...  Rating 5 out of 5   
5207  ['Wifi', 'Cable TV', 'Laptop friendly workspac...  Rating 5 out of 5   
5519  ['Wifi', 'Cable TV', 'Laptop friendly workspac...  Rating 5 out of 5  

By looking at the data, I noticed that this sample representing shared room in hostle, and they did not provide any information about the number of bathrooms. For simplicity, I will assume that we have only "1 shared bathroom"! 

In [125]:
for i in range(len(BangkokData)): 
    if(BangkokData['Bathrooms_num'][i]== 'Awesome place! Laundry service was available and affordable, a well equipped kitchen was provided, and the bathrooms and rooms were clean.'):
        BangkokData['Bathrooms_num'][i]= '1 shared bath'
        
    

C:\Users\KBrig\AnacondaNada\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy



In [126]:
print(BangkokData['Bathrooms_num'].unique())

['1 bath' '1 private bath' '2 baths' '1.5 baths' '0 baths' '1 shared bath'
 '2 shared baths' '4 shared baths' '1.5 private baths' '8 shared baths'
 '1.5 shared baths' '3 baths' '4 baths' '2.5 baths' '3.5 baths' nan
 '4.5 baths' '6 shared baths' '4.5 shared baths']


In [127]:
BangkokData['Bathrooms_num_int'] = bathrooms_int(BangkokData)

In [148]:
for i in range(10):
    print(BangkokData['Bathrooms_num'][i],BangkokData['Bathrooms_num_int'][i])

1 bath 1.0
1 bath 1.0
1 bath 1.0
1 bath 1.0
1 bath 1.0
1 bath 1.0
1 private bath 1.0
1 bath 1.0
1 bath 1.0
1 bath 1.0


### Shared vs. Private Bathroom

In [153]:
BangkokData['shared_bathroom']= shared_bathroom(BangkokData)

In [156]:
BangkokData[['Bathrooms_num','shared_bathroom']][0:10]

Bathrooms_num  shared_bathroom
0          1 bath            False
1          1 bath            False
2          1 bath            False
3          1 bath            False
4          1 bath            False
5          1 bath            False
6  1 private bath            False
7          1 bath            False
8          1 bath            False
9          1 bath            False

### Number of Bedrooms 

In [53]:
print(type(BangkokData['Bedrooms_no'][0]))
print(BangkokData['Bedrooms_no'].unique())

<class 'str'>
['1 bedroom' nan
 'Studio accommodation with nice facilities is yours. Facilities are as follows: One king bedroom 2-3 walking distance to Thong Lo Station station'
 'We were very happy when we first booked with Jackrit as the condo had what we were looking for. However, the toilet was an utter disappointment. It had this strong, unbearable odor that permeated through to the bedroom. Even though Jackrit had someone look at it, sadly, it did…Read more'
 'G2015 Single bedroom E1 at Sukhumvit soi64.near sky trian (BTS) Punnawithi station.Room size 26sqm.Have 1 kitchen,1 bathroom and 1 balcony.'
 "Its a nice condo about 12 mins walk to the ON NUT BTS, the condo is on the 8th top floor with good views on BKK city and the pool / condo courtyard,I stayed in another condo after this and whist the other was a one bedroom unit instead of a studio Ho's place and support was…Read more"
 '2 bedrooms' 'New & modern one-bedroom suite near BTS Phyathai'
 'Clean Cozy Condo 1bedroom BTS ba

### Number of Beds

In [39]:
print(type(BangkokData['Beds_no'][0]))
print(BangkokData['Beds_no'].unique())

<class 'str'>
['1 bed' '2 beds' '3 beds' nan '4 beds' '0 beds'
 'PRIVATE ROOM IN BED AND BREAKFAST' '5 beds' '6 beds' '7 beds' '8 beds'
 'Private room in bed and breakfast']


In [40]:
BangkokData['Beds_int'] = float(0)

In [41]:
BangkokData['Beds_int']= beds_int(BangkokData)

In [42]:
for i in range(10):
    print(BangkokData['Beds_no'][i],BangkokData['Beds_int'][i])

1 bed 1.0
1 bed 1.0
2 beds 2.0
1 bed 1.0
3 beds 3.0
3 beds 3.0
1 bed 1.0
1 bed 1.0
1 bed 1.0
1 bed 1.0


### Latitude

In [191]:
print(type(BangkokData['Latitude'][0]))

<class 'numpy.float64'>


### Longitude

In [192]:
print(type(BangkokData['Longitude'][0]))

<class 'numpy.float64'>


# Cairo

In [166]:
CairoData.columns


Index(['CheckIn', 'CheckOut', 'Has_Weekend', 'Title', 'Price', 'Super_host',
       'Latitude', 'Longitude', 'EntireHomeVsRoom', 'Bedrooms_no', 'Beds_no',
       'Bathrooms_num', 'Sleeping_engagment', 'Amenities', 'Accuracy_stars',
       'Communication_stars', 'Cleanliness_stars', 'Location_stars',
       'CheckIn_stars', 'Value_stars', 'Reviewers_no', 'Review'],
      dtype='object')

### Does it have a weekend?

In [167]:
CairoData['Has_Weekend'].unique()

array([ True])

In [168]:
CairoData.drop(['Has_Weekend','Reviewers_no'],axis=1, inplace=True)

In [169]:
CairoData.columns

Index(['CheckIn', 'CheckOut', 'Title', 'Price', 'Super_host', 'Latitude',
       'Longitude', 'EntireHomeVsRoom', 'Bedrooms_no', 'Beds_no',
       'Bathrooms_num', 'Sleeping_engagment', 'Amenities', 'Accuracy_stars',
       'Communication_stars', 'Cleanliness_stars', 'Location_stars',
       'CheckIn_stars', 'Value_stars', 'Review'],
      dtype='object')

### Is Super Host?

In [170]:
CairoData.Super_host[0:10]

0                       NaN
1                       NaN
2                       NaN
3                       NaN
4                       NaN
5                       NaN
6                       NaN
7     Bassem is a Superhost
8                       NaN
9    Mohamed is a Superhost
Name: Super_host, dtype: object

In [171]:
CairoData['Super_host'] = super_host(CairoData)

In [172]:
CairoData.Super_host[0:10]

0    0
1    0
2    0
3    0
4    0
5    0
6    0
7    1
8    0
9    1
Name: Super_host, dtype: int64

### Price

In [173]:
CairoData['Price'] = price_format(CairoData)

In [174]:
print(type(CairoData['Price'][0]))
print(CairoData['Price'][0:10])

<class 'numpy.float64'>
0    15.0
1    12.0
2    22.0
3    19.0
4    23.0
5    25.0
6    40.0
7    37.0
8    65.0
9    11.0
Name: Price, dtype: float64


### Number of Bathrooms

In [175]:
print(type(CairoData['Bathrooms_num'][0]))
print(CairoData['Bathrooms_num'].unique())

<class 'str'>
['2 shared baths' '1.5 baths' '1 shared bath' '1.5 private baths'
 '1 private bath' '1 bath' '2 baths' '3 baths' '12 private baths'
 '8 private baths' '1.5 shared baths' '2.5 baths' nan '4 baths'
 '2 private baths' '0 shared baths' '3 private baths' '3 shared baths'
 '10 private baths' '5 private baths' '6 baths' '4 private baths'
 '3.5 baths' '4 shared baths' '25 private baths' '20 private baths']


In [176]:
CairoData['Bathrooms_num_int']= float(0)

In [179]:
CairoData['Bathrooms_num_int']= bathrooms_int(CairoData) 

c:\users\kbrig\airbnbrepo\airbnb\bnb\data_processing\clean.py:17: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy

c:\users\kbrig\airbnbrepo\airbnb\bnb\data_processing\clean.py:13: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy



In [180]:
for i in range(10):
    print(CairoData['Bathrooms_num'][i],CairoData['Bathrooms_num_int'][i])

2 shared baths 2.0
1.5 baths 1.5
1 shared bath 1.0
1.5 private baths 1.5
1 shared bath 1.0
1 private bath 1.0
1 private bath 1.0
1.5 baths 1.5
1 bath 1.0
1 shared bath 1.0


### Shared vs. Private Bathrooms

In [181]:
CairoData['shared_bathroom']= shared_bathroom(CairoData)

In [183]:
CairoData[['Bathrooms_num','shared_bathroom']][0:10]

Bathrooms_num  shared_bathroom
0     2 shared baths             True
1          1.5 baths            False
2      1 shared bath             True
3  1.5 private baths            False
4      1 shared bath             True
5     1 private bath            False
6     1 private bath            False
7          1.5 baths            False
8             1 bath            False
9      1 shared bath             True

### Number of Bedrooms

In [184]:
print(type(CairoData['Bedrooms_no'][0]))
print(CairoData['Bedrooms_no'].unique())
print(CairoData['Bedrooms_no'][0:5])

<class 'str'>
['1 bedroom' '2 bedrooms'
 'This fully furnished studio garden apartment includes 1 bedroom, 2 reception units, 1 bathroom..'
 'Expat apartment - Clean and cosy bedroom' nan
 'Cairo Zahraa El Maadi, one bedroom.'
 'Paradise at the Point! This luxurious 1 bedroom, 1 bathroom bath modern-built estate on the Open view is a dream home in a high-end community.'
 '20 bedrooms' '3 bedrooms'
 'Double bedroom in Paris Hotel Breakfast included .'
 'Cozy queen bedroom, perfect location.'
 'Cozy,clean apartment located in Maadi, one bedroom'
 '2 Single bedroom in Paris Hotel Breakfast included'
 'Apartment(3bedroom) Eltagamo new cairo - 4' '4 bedrooms'
 'Private bedroom in a cosy apartment' '10 bedrooms'
 'Apartment(2bedroom) Eltagamo new cairo'
 'One bedroom with garden in The Village, New Cairo.' '6 bedrooms'
 '15 bedrooms' 'Apartment(3bedroom) Eltagamo new cairo - 5'
 'Apartment(3bedroom) Eltagamo new cairo - 3'
 'Princess Apartment 3 bedrooms' 'Beautiful 2 bedrooms close to down 

### Number of Beds

In [43]:
print(type(CairoData['Beds_no'][0]))
print(CairoData['Beds_no'].unique())

<class 'str'>
['2 beds' '1 bed' '4 beds' nan 'Private room in bed and breakfast'
 '3 beds' '26 beds' '6 beds' '5 beds' '8 beds'
 'Stylish brand new fully furnished studio apartment with queen size bed and open plan kitchen. kitchen is fully equipped with all cutlery and crockery, washing machine, cooker, microwave, kettle, toaster. bed linen and towels are provided.'
 '16 beds' '50 beds' '37 beds' '7 beds' '10 beds' '18 beds' '30 beds']


In [47]:
CairoData[CairoData['Beds_no']== 'Stylish brand new fully furnished studio apartment with queen size bed and open plan kitchen. kitchen is fully equipped with all cutlery and crockery, washing machine, cooker, microwave, kettle, toaster. bed linen and towels are provided.']

CheckIn         CheckOut  Has_Weekend  \
70    2019-03-14 0:00  2019-03-19 0:00         True   
557   2019-03-28 0:00  2019-04-02 0:00         True   
896   2019-04-11 0:00  2019-04-16 0:00         True   
902   2019-04-11 0:00  2019-04-16 0:00         True   
1211  2019-04-25 0:00  2019-04-30 0:00         True   
2102  2019-06-06 0:00  2019-06-11 0:00         True   
2380  2019-06-20 0:00  2019-06-25 0:00         True   
2681  2019-07-04 0:00  2019-07-09 0:00         True   
3001  2019-07-18 0:00  2019-07-23 0:00         True   
3317  2019-08-01 0:00  2019-08-06 0:00         True   
3633  2019-08-15 0:00  2019-08-20 0:00         True   
3920  2019-08-29 0:00  2019-09-03 0:00         True   
4169  2019-09-12 0:00  2019-09-17 0:00         True   
4784  2019-10-10 0:00  2019-10-15 0:00         True   
5095  2019-10-24 0:00  2019-10-29 0:00         True   
5309  2019-11-07 0:00  2019-11-12 0:00         True   
5614  2019-11-21 0:00  2019-11-26 0:00         True   
5917  2019-12-05 0:00  2019-12-10 0:00         True   
6212  2019-12-19 0:00  2019-12-24 0:00         True   
6510  2020-01-02 0:00  2020-01-07 0:00         True   
6816  2020-01-16 0:00  2020-01-21 0:00         True   
7119  2020-01-30 0:00  2020-02-04 0:00         True   
7423  2020-02-13 0:00  2020-02-18 0:00         True   
7725  2020-02-27 0:00  2020-03-03 0:00         True   

                                                  Title Price Super_host  \
70    fully furnished apt in the village palm hills,...   $63        NaN   
557   fully furnished apt in the village palm hills,...   $63        NaN   
896   fully furnished apt in the village palm hills,...   $63        NaN   
902   fully furnished apt in the village palm hills,...   $63        NaN   
1211  fully furnished apt in the village palm hills,...   $63        NaN   
2102  fully furnished apt in the village palm hills,...   $63        NaN   
2380  fully furnished apt in the village palm hills,...   $63        NaN   
2681  fully furnished apt in the village palm hills,...   $63        NaN   
3001  fully furnished apt in the village palm hills,...   $63        NaN   
3317  fully furnished apt in the village palm hills,...   $69        NaN   
3633  fully furnished apt in the village palm hills,...   $70        NaN   
3920  fully furnished apt in the village palm hills,...   $71        NaN   
4169  fully furnished apt in the village palm hills,...   $77        NaN   
4784  fully furnished apt in the village palm hills,...   $78        NaN   
5095  fully furnished apt in the village palm hills,...   $78        NaN   
5309  fully furnished apt in the village palm hills,...   $78        NaN   
5614  fully furnished apt in the village palm hills,...   $78        NaN   
5917  fully furnished apt in the village palm hills,...   $78        NaN   
6212  fully furnished apt in the village palm hills,...   $78        NaN   
6510  fully furnished apt in the village palm hills,...   $78        NaN   
6816  fully furnished apt in the village palm hills,...   $78        NaN   
7119  fully furnished apt in the village palm hills,...   $78        NaN   
7423  fully furnished apt in the village palm hills,...   $78        NaN   
7725  fully furnished apt in the village palm hills,...   NaN        NaN   

      Latitude  Longitude  EntireHomeVsRoom Bedrooms_no   ...     \
70    30.01949   31.50053  Entire apartment   1 bedroom   ...      
557   30.01949   31.50053  Entire apartment   1 bedroom   ...      
896   30.01949   31.50053  Entire apartment   1 bedroom   ...      
902   30.01949   31.50053  Entire apartment   1 bedroom   ...      
1211  30.01949   31.50053  Entire apartment   1 bedroom   ...      
2102  30.01949   31.50053  Entire apartment   1 bedroom   ...      
2380  30.01949   31.50053  Entire apartment   1 bedroom   ...      
2681  30.01949   31.50053  Entire apartment   1 bedroom   ...      
3001  30.01949   31.50053  Entire apartment   1 bedroom   ...      
3317  30.01949   31.50053  Entire apartment   1 bedroom  

In [48]:
for i in range(len(BangkokData)): 
    if(CairoData['Beds_no'][i]== 'Stylish brand new fully furnished studio apartment with queen size bed and open plan kitchen. kitchen is fully equipped with all cutlery and crockery, washing machine, cooker, microwave, kettle, toaster. bed linen and towels are provided.'):
        CairoData['Beds_no'][i]= '1 bed'

C:\Users\KBrig\AnacondaNada\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy



In [49]:
CairoData['Beds_int']= float(0)

In [50]:
CairoData['Beds_int']= beds_int(CairoData)

c:\users\kbrig\airbnbrepo\airbnb\bnb\data_processing\clean.py:29: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy

c:\users\kbrig\airbnbrepo\airbnb\bnb\data_processing\clean.py:23: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy

c:\users\kbrig\airbnbrepo\airbnb\bnb\data_processing\clean.py:25: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy



In [51]:
for i in range(10):
    print(CairoData['Beds_no'][i],CairoData['Beds_int'][i])

2 beds 2.0
1 bed 1.0
1 bed 1.0
2 beds 2.0
1 bed 1.0
1 bed 1.0
1 bed 1.0
2 beds 2.0
1 bed 1.0
1 bed 1.0


### Latitude

In [189]:
print(type(CairoData['Latitude'][0]))

<class 'numpy.float64'>


### Longitude

In [190]:
print(type(CairoData['Longitude'][0]))

<class 'numpy.float64'>


## Categorical Data

### NewYork

In [54]:
print(NewYorkData['Accuracy_stars'].unique())


['Rating 5 out of 5' 'Rating 4.5 out of 5' nan 'Rating 4 out of 5'
 'Rating 2.5 out of 5' 'Rating 3.5 out of 5']


In [22]:
type(NewYorkData['Accuracy_stars'][0])

str

In [23]:
t = num_of_stars(NewYorkData['Accuracy_stars'])

c:\users\kbrig\airbnbrepo\airbnb\bnb\data_processing\clean.py:45: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy



AttributeError: 'float' object has no attribute 'split'

In [60]:
t.split(' ')[1]

'5'

### Bangkok

### Cairo

### Price Analysis

### NewYork

In [ ]:
AvgPrice_NY = group_by_date(NewYorkData) 

In [ ]:
#reate a data object
price_data = go.Scatter(x=AvgPrice_NY.index,y=AvgPrice_NY.values)

#set up the plot layout
layout1 = go.Layout(title='NY_ Average Price Plot', xaxis=dict(title='CheckIn_Date'),yaxis=dict(title='Price'))

# create a figure and display it
fig1 = go.Figure(data=[price_data],layout=layout1)


#display the plot interactively
iplot(fig1)


### Bangkok

In [ ]:
AvgPrice_Bngkok = group_by_date(BangkokData) 
AvgPrice_Bngkok

In [ ]:
#reate a data object
price_data_Bngkok = go.Scatter(x=AvgPrice_Bngkok.index,y=AvgPrice_Bngkok.values)
#set up the plot layout
layout2 = go.Layout(title='Bangkok_ Average Price Plot', xaxis=dict(title='CheckIn_Date'),yaxis=dict(title='Price'))
# create a figure and display it
fig2 = go.Figure(data=[price_data_Bngkok],layout=layout2)
#display the plot interactively
iplot(fig2)


### Cairo

In [ ]:

AvgPrice_Cairo = group_by_date(CairoPrice)
AvgPrice_Cairo

In [ ]:
#reate a data object
price_data_Cairo = go.Scatter(x=AvgPrice_Cairo.index,y=AvgPrice_Cairo.values)

#set up the plot layout
layout3 = go.Layout(title='Cairo_ Average Price Plot', xaxis=dict(title='CheckIn_Date'),yaxis=dict(title='Price'))

# create a figure and display it
fig3 = go.Figure(data=[price_data_Cairo],layout=layout3)

#display the plot interactively
iplot(fig3)

